core/src/test/scala/com/microsoft/azure/synapse/ml/exploratory/DistributionBalanceMeasureSuite.scala

```diff
@@ -105,4 +112,131 @@ class DistributionBalanceMeasureSuite extends DataBalanceTestBase with Transform
+    assert(actual(JSDISTANCE) === expected.JSDISTANCE)
+    assert(actual(INFNORMDISTANCE) === expected.INFNORMDISTANCE)
+    assert(actual(TOTALVARIATIONDISTANCE) === expected.TOTALVARIATIONDISTANCE)
+    assert(actual(WASSERSTEINDISTANCE) === expected.WASSERSTEINDISTANCE)
+    assert(actual(CHISQUAREDTESTSTATISTIC) === expected.CHISQUAREDTESTSTATISTIC)
+    assert(actual(CHISQUAREDPVALUE) === expected.CHISQUAREDPVALUE)
+  }
+
+  private def actualCustomDistFeature2: Map[String, Double] =
+    METRICS zip actualCustomDist.filter(col("FeatureName") === feature2)
+      .select(array(col("DistributionBalanceMeasure.*")))
+      .as[Array[Double]]
+      .head toMap
+
+  private def expectedCustomDistFeature2 = getFeatureStats(sensitiveFeaturesDf.groupBy(feature2))
+    .select(feature2, featureProbCol, featureCountCol)
+    .as[(String, Double, Double)].collect()
+
+  private object ExpectedCustomDistFeature2 {
+    // Values were computed using:
+    // val (numRows, numFeatures) = (sensitiveFeaturesDf.count.toDouble, expectedCustomDistFeature2.length)
+    // val (featureValues, obsProbs, obsCounts) = expectedCustomDistFeature2.unzip3
+    // val refProbs = featureValues.map(customDistribution.get(1).getOrDefault(_, 0.0)) // idx 1 = Ethnicity
+    // val refCounts = refProbs.map(_ * numRows)
+    // val CALC = DistributionMetricsCalculator(refProbs, refCounts, obsProbs, obsCounts, numFeatures)
+    val KLDIVERGENCE = Double.PositiveInfinity // ln(obsProb(Other) / refProb(Other)) = ln(0.111 / 0.0) = +Inf
+    val JSDISTANCE = Double.NaN // ln(refProb(Other) / avgRefAObsProb(Other)) = ln(0.0 / 0.055) = -Inf -> sum(...) = NaN
+    val INFNORMDISTANCE = 0.1111111111111111
+    val TOTALVARIATIONDISTANCE = 0.11277777777777775
+    val WASSERSTEINDISTANCE = 0.05638888888888888
+    // (obsCount(Other) - refCount(Other))^2 / refCount(Other)) = (1 - 0)^2 / 0 = 1 / 0 = +Inf
+    val CHISQUAREDTESTSTATISTIC = Double.PositiveInfinity
+    // Since Chi^2 represents the confidence that x <= Y and Y is +inf, p-value = 1.0 - confidence = 1.0 - 1.0 = 0.0
+    val CHISQUAREDPVALUE = 0
+  }
+
+  test(s"DistributionBalanceMeasure can a custom reference distribution with missing values ($feature2)") {
+    // The custom reference distribution for Ethnicity is Map("Asian" -> 0.33, "White" -> 0.33, "Black" -> 0.33)
+    // This does NOT include all unique values in the dataframe being transformed; 'Other' is left out
+    // which means that it should default to a reference probability of 0.00
+    val actual = actualCustomDistFeature2
+    val expected = ExpectedCustomDistFeature2
+
+    assert(actual(INFNORMDISTANCE) === expected.INFNORMDISTANCE)
+    assert(actual(TOTALVARIATIONDISTANCE) === expected.TOTALVARIATIONDISTANCE)
+    assert(actual(WASSERSTEINDISTANCE) === expected.WASSERSTEINDISTANCE)
+
+    // TODO: The following measures are handling a probability distribution of 0.0 differently than expected.
```

Comment:
I wanted to call out this TODO. Janhavi and I are investigating what the behavior should be, and we were thinking to make a fix after this PR so that she can start integrating this in pyspark.

I'm currently doing an investigation of the bugs happening with KL Div + JS Dist + Chi^2 when using 0.0 ref probability distribution and I'll share my findings soon.

Let me know your thoughts :)

Reply:
KL Divergence is infinite in certain cases where prob is 0.0 on one dist and >0.0 on another dist.

Reply:
TY for that callout! You are right, in this case it should be infinity. After adding edge cases for calculating entropy and chi^2, the measure outputs are the same as scipy outputs. I removed the TODO and marking this as resolved